### PR TITLE
docs(cookie-session): use async

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: docs/source/en/core/cookie-and-session.md [hunk 1]
```diff
@@ -115,20 +118,23 @@ So the concept of Session, which is built on top of Cookie,
...
-    success: true,
-    posts,
-  };
-};
+class HomeController extends Controller {
+  async fetchPosts() {
+    const ctx = this.ctx;
+    // get content from session
+    const userId = ctx.session.userId;
+    const posts = yield ctx.service.post.fetch(userId);
```
Comment (popomore) [inline docs/source/en/core/cookie-and-session.md]: await

File: docs/source/en/core/cookie-and-session.md [hunk 2]
```diff
@@ -175,13 +181,14 @@ To config it, you can simply set `app.sessionStore`.
 // app.js
 module.exports = app => {
   app.sessionStore = {
-    * get (key) {
+    // support promise / async
+    get (key) {
```
Comment (popomore) [inline docs/source/en/core/cookie-and-session.md]: 不是改成 async？

Comment (atian25) [reply docs/source/en/core/cookie-and-session.md]: promise base 的，这里应该支持 async 也支持 () => Promise 吧，下面几个都没，所以就加个注释。

Comment (atian25) [reply docs/source/en/core/cookie-and-session.md]: fixed

File: docs/source/en/core/cookie-and-session.md [hunk 3]
```diff
@@ -241,19 +250,19 @@ By default, if user requests don't result in modification of Session,
...
 
 For example, we create a middleware in the application.
 It forces saving session in every request, in order to extend session's expiration time.
 
 ```js
 // app/middleware/save_session.js
 module.exports = () => {
-  return function* (next) {
-    yield next;
+  return async saveSession(ctx, next) {
```
Comment (dead-horse) [inline docs/source/en/core/cookie-and-session.md]: async function saveSession

File: docs/source/zh-cn/core/cookie-and-session.md [hunk 4]
```diff
@@ -176,36 +181,38 @@ exports.sessionRedis = {
...
+默认情况下，当用户请求没有导致 Session 被修改时，框架都不会延长 Session 的有效期，但是在有些场景下，我们希望用户每次访问都刷新 Session 的有效时间，这样用户只有在长期未访问我们的网站的时候才会被登出。这个功能我们可以通过 `ctx.session.save()` 来实现。
 
 例如，我们在项目中增加一个中间件，让它在 Session 有值的时候强制保存一次，以达到延长 Session 有效期的目的。
 
 ```js
 // app/middleware/save_session.js
 module.exports = () => {
-  return function* (next) {
-    yield next;
+  return async saveSession(ctx, next) {
```
Comment (dead-horse) [inline docs/source/zh-cn/core/cookie-and-session.md]: 同上，其他 +1，改完合并吧

Comment (atian25) [reply docs/source/zh-cn/core/cookie-and-session.md]: fixed

**Pull request merged**